### DISCUSSION TITLE
Question: Index Axis without Borrowing?

### DISCUSSION THREAD

**Discussion by skailasa** [rust-ndarray/ndarray#1008] [Q&A] [answered]
How does one idiomatically index elements of an array instance without borrowing the array? Can one just clone the given elements indexed? For example,

```rust
let column = matrix.index_axis(ndarray::Axis(1), 0);
```
takes ownership of the 'matrix' variable. How can I use matrix again, without cloning it? Again, a fairly noob question from a novice rust user!

**Comment by bluss** [comment, marked as answer]
`index_axis` doesn't take ownership of matrix at all. Borrows, but does not take it. Check out all the methods called `index_axis`, the slicing methods and the general docs for them.
